Riot/Modules/Authentication/LegacyAuthenticationCoordinator.swift

```diff
@@ -141,119 +145,49 @@ final class AuthenticationCoordinator: NSObject, AuthenticationCoordinatorProtoc
-                                }
-                            } else {
-                                // Try to setup cross-signing without authentication parameters in case if a grace period is enabled
-                                self.crossSigningService.setupCrossSigningWithoutAuthentication(for: session) {
-                                    MXLog.debug("[AuthenticationCoordinator] sessionStateDidChange: Bootstrap succeeded without credentials")
-                                    self.authenticationDidComplete()
-                                } failure: { error in
-                                    MXLog.error("[AuthenticationCoordinator] sessionStateDidChange: Do not know how to bootstrap cross-signing. Skip it.")
-                                    crypto.setOutgoingKeyRequestsEnabled(true, onComplete: nil)
-                                    self.authenticationDidComplete()
-                                }
-                            }
-                        } else {
-                            crypto.setOutgoingKeyRequestsEnabled(true, onComplete: nil)
-                            self.authenticationDidComplete()
-                        }
-                    case .crossSigningExists:
-                        guard self.canPresentAdditionalScreens else {
-                            MXLog.debug("[AuthenticationCoordinator] sessionStateDidChange: Delaying presentCompleteSecurity during onboarding.")
-                            self.isWaitingToPresentCompleteSecurity = true
-                            return
-                        }
-                        
-                        MXLog.debug("[AuthenticationCoordinator] sessionStateDidChange: Complete security")
-                        self.presentCompleteSecurity()
-                    default:
-                        MXLog.debug("[AuthenticationCoordinator] sessionStateDidChange: Nothing to do")
-                        
-                        crypto.setOutgoingKeyRequestsEnabled(true, onComplete: nil)
-                        self.authenticationDidComplete()
-                    }
-                } failure: { [weak self] error in
-                    MXLog.error("[AuthenticationCoordinator] sessionStateDidChange: Fail to refresh crypto state with error: \(error)")
-                    crypto.setOutgoingKeyRequestsEnabled(true, onComplete: nil)
-                    self?.authenticationDidComplete()
-                }
-            } else {
-                authenticationDidComplete()
-            }
-        }
-    }
 }
 
 // MARK: - AuthenticationViewControllerDelegate
-extension AuthenticationCoordinator: AuthenticationViewControllerDelegate {
+extension LegacyAuthenticationCoordinator: AuthenticationViewControllerDelegate {
     func authenticationViewController(_ authenticationViewController: AuthenticationViewController!, didLoginWith session: MXSession!, andPassword password: String!) {
```

Comment:
Good idea. I've made `password` optional in another branch but wasn't 100% sure to reason about the `MXSession`. I'll add the check in that branch so as to avoid conflicts.